### DISCUSSION TITLE
Is friendly-snippets a necessary dependence for luasnip?

### DISCUSSION THREAD

**Discussion by prexhu** [L3MON4D3/LuaSnip#1215] [Q&A] [answered]
I's wonder that is `friendly-snipppets` a necessary dependence for luasnip?
I use lazyvim as my machine neovim starter. I follow the LuaSnip repo guide to finish the plugin installation. I add some test tex-snips in the folder `~/.config/nvim/lua/plugin/snippets`. I can not get the snips prompt while I edit a tex file(I try the command `LuaSnipListActive`, and it shows snippets). And I mention this warning in `friend-snippets` repo.
![image](https://github.com/user-attachments/assets/c0dc3c94-a883-4167-be62-98b519776051)
(I not install `friend-snippets` on machine yet, because I do not wanna use vscode_snips, which might be the reason for luasnip doesn't work on my machine.)

This situation is what I have met.
![image](https://github.com/user-attachments/assets/2627ee9d-244b-4ede-97a2-c2ec90034e0d)

And this is what I wanna get(with snippets prompt)
![image](https://github.com/user-attachments/assets/f6ac8254-01e4-4d1e-87b5-820bf800bc23)

(I am a beginner at computer 🤯, and I feel like I seem stupid in this scope frequently, forgive me for my ignorance.)

**Comment by L3MON4D3** [comment, marked as answer]
Hi :)

Luasnip itself does not add the prompt, we only handle manual expansion (so by typing the snippet trigger and pressing whatever key you've bound `require("luasnip").expand` to), for the list you need `cmp_luasnip`.
The pictures you show look more like a lsp-server with snippets disabled on the top, and one with snippets enabled on the bottom, but that's unrelated to expanding your own snippets.

**Reply by prexhu** [reply]
yeah, I have already solved my problem by adding source `luasnip` to `nvim-cmp` configuration, thank you very much!

**Reply by L3MON4D3** [reply]
Ah, perfect :ok_hand:

**Comment by ChenZhongPu** [comment]
I am using `luasnip` in `lazy.nvim` with `friendly-snippet` as a dependency. But the self-defined snippets cannot be loaded, while the built-in snippets can be loaded correctly. How to solve it? I also posted a discussion [here](https://github.com/LazyVim/LazyVim/discussions/4921).

**Reply by L3MON4D3** [reply]
Are you calling `require("luasnip.loaders.from_vscode").lazy_load(paths = {"<path-to-snippets-dir>"})` (or `.load(...)`)?

**Reply by ChenZhongPu** [reply]
This calling is involved by its [setting](https://www.lazyvim.org/extras/coding/luasnip):

```lua
{
  "rafamadriz/friendly-snippets",
  config = function()
    require("luasnip.loaders.from_vscode").lazy_load()
    require("luasnip.loaders.from_vscode").lazy_load({ paths = { vim.fn.stdpath("config") .. "/snippets" } })
  end,
}
```

**Reply by L3MON4D3** [reply]
Ah, that looks correct :+1:
Could you add a `require("luasnip").log.set_loglevel("info")` beforeany lazy_load-calls and check the log after the load-calls (`require("luasnip").log.open()`), and check for references to your `package.json` or `snippets`-directory?

**Reply by ChenZhongPu** [reply]
The settings above are in fact a part of another plugin [LazyVim](https://github.com/LazyVim/LazyVim/blob/475e3f32b82db0cc497f712953993dcce4f048c6/lua/lazyvim/plugins/extras/coding/luasnip.lua#L15). Then how can I add the log level in my own config?

**Reply by ChenZhongPu** [reply]
My fault. It is caused by LazyVim configs, and this bug was [fixed](https://github.com/LazyVim/LazyVim/pull/4987/files) 15 hours ago.
